src/main/java/mat/server/service/impl/MatContextServiceUtil.java

```diff
@@ -11,17 +14,32 @@
 import mat.model.clause.MeasureShareDTO;
 import mat.model.clause.ShareLevel;
 import mat.model.cql.CQLLibraryShareDTO;
+import mat.server.FeatureFlagServiceImpl;
 import mat.server.LoggedInUserUtil;
 import mat.shared.model.util.MeasureDetailsUtil;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
-public class MatContextServiceUtil {
+@Service
+public class MatContextServiceUtil implements InitializingBean {
 
     private static BigDecimal QDM_VER_BEFORE_CONVERSION = new BigDecimal("5.5");
     private static BigDecimal MAT_VER_BEFORE_CONVERSION = new BigDecimal("5.8");
 
     private boolean isMeasure;
 
-    private static MatContextServiceUtil instance = new MatContextServiceUtil();
+    private static MatContextServiceUtil instance;
+
+    @Autowired
+    FeatureFlagServiceImpl featureFlag;
+
+    @Override
+    public void afterPropertiesSet() {
+        instance = this;
+    }
+
+    private Map<String, Boolean> featureFlagMap = new HashMap<>();

```

Comment:
why do you store it here? if it's a singleton then it's not thread-safe.
also, it's not for caching because it's always refreshed..
could you explain?

Reply:
I removed it